src/fec_context.h

```diff
@@ -70,123 +84,98 @@ class DecodeContext {
+        inv_A_i->zero_fill();
+        S->zero_fill();
 
-        poly1_n->zero();
-        poly2_n->zero();
+        vec1_n->zero_fill();
+        vec2_n->zero_fill();
     }
 
-    ~DecodeContext()
-    {
-        if (A)
-            delete A;
-        if (A_fft_2k)
-            delete A_fft_2k;
-        if (inv_A_i)
-            delete inv_A_i;
-        if (S)
-            delete S;
-        if (poly1_n)
-            delete poly1_n;
-        if (poly2_n)
-            delete poly2_n;
-    }
+    ~DecodeContext() {}
 
-    unsigned get_len_2k()
+    unsigned get_len_2k() const
     {
         return len_2k;
     }
 
-    vec::Poly<T>* get_A()
-    {
-        return A;
-    }
-
-    vec::Poly<T>* get_A_fft_2k()
-    {
-        return A_fft_2k;
-    }
-
-    vec::Poly<T>* get_inv_A_i()
-    {
-        return inv_A_i;
-    }
-
-    vec::Poly<T>* get_poly1_n()
-    {
-        return poly1_n;
-    }
-
-    vec::Poly<T>* get_poly2_n()
-    {
-        return poly2_n;
-    }
-
-    vec::Poly<T>* get_poly1_2k()
-    {
-        return poly1_2k;
-    }
-
-    vec::Poly<T>* get_poly2_2k()
-    {
-        return poly2_2k;
-    }
-
-    vec::Poly<T>* get_S()
-    {
-        return S;
-    }
-
-    vec::Vector<T>* get_frag_ids()
+    vec::Vector<T>* get_fragments_id() const
     {
         return fragments_ids;
     }
 
-    void set_frag_ids(vec::Vector<T>* fragments_ids)
-    {
-        this->fragments_ids = fragments_ids;
-    }
-
-    int get_vx_zero()
-    {
-        return vx_zero;
+    vec::Vector<T>* get_vector(unsigned vec_type) const
+    {
+        switch (vec_type) {
+        case vec_A_fft_2k:
+            return A_fft_2k.get();
+        case vec_inv_A_i:
+            return inv_A_i.get();
+        case vec_n_1:
+            return vec1_n.get();
+        case vec_n_2:
+            return vec2_n.get();
+        case vec_2k_1:
+            return vec1_2k.get();
+        case vec_2k_2:
+            return vec2_2k.get();
+        default:
+            return nullptr;
+        }
     }
 
-    void set_vx_zero(int value)
+    vec::Poly<T>* get_poly(unsigned vec_type) const
     {
-        this->vx_zero = value;
+        switch (vec_type) {
+        case vec_A:
+            return A.get();
+        case vec_S:
+            return S.get();
+        default:
+            return nullptr;
+        }
     }
 
-    void init(vec::Vector<T>* fragments_ids, const vec::Vector<T>& vx)
+    void init(
+        vec::Vector<T>* fragments_ids,
+        const vec::Vector<T>& vx,
+        bool nf4 = false)
     {
         this->fragments_ids = fragments_ids;
 
         // compute A(x) = prod_j(x-x_j)
-        A->set(0, 1);
+        if (nf4)
+            A->set(0, this->gf->replicate(1));
+        else
+            A->set(0, 1);
         for (int i = 0; i < k; ++i) {
             A->mul_to_x_plus_coef(this->gf->sub(0, vx.get(i)));
         }
 
         // compute A'(x) since A_i(x_i) = A'_i(x_i)
         vec::Poly<T> _A(*A);
-        _A.derivative();
+        if (nf4)
```

Comment:
Add braces {}, even for single lines (the linter don't enforce it yet, but as we enable more lint check this will probably becomes mandatory).
Ditto for the lines above (162-165)